pkg/querier/cardinality_analysis_handler.go

```diff
@@ -101,8 +102,12 @@ func ActiveSeriesCardinalityHandler(distributor Distributor, limits *validation.
 			return
 		}
 
-		res, err := distributor.ActiveSeries(ctx, req.Matchers)
+		res, err := d.ActiveSeries(ctx, req.Matchers)
 		if err != nil {
+			if errors.Is(err, distributor.ErrResponseTooLarge) {
+				http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
This should probably be a different status code so the client knows how to resolve the failure (i.e. increase the shard count in the request), but I'm not aware of a good candidate.

Reply:
How about this?

[413 Payload Too Large](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413)
Request entity is larger than limits defined by server. The server might close the connection or return an Retry-After header field.

Reply:
We are already using 413 for the case response size exceeds limit ([here](https://github.com/grafana/mimir/blob/8aa3a5a4e8a3614a3108cd05c0ade96fffde735b/pkg/querier/worker/scheduler_processor.go#L265)), so I'll change this PR to do the same.